client/extensions/woocommerce/components/form-dimensions-input/index.jsx

```diff
@@ -1,56 +1,112 @@
 
 /**
  * Internal dependencies
  */
 import FormTextInput from 'components/forms/form-text-input';
 import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
 
-function FormDimensionsInput( {
-	className,
-	dimensions,
-	unit,
-	onChange,
-	translate,
-	noWrap,
-} ) {
-	const classes = classNames( 'form-dimensions-input', className, { 'no-wrap': noWrap } );
-
-	return (
-		<div className={ classes }>
-			<FormTextInput
-				name="length"
-				placeholder={ translate( 'L' ) }
-				type="number"
-				value={ dimensions && dimensions.length || '' }
-				onChange={ onChange }
-				className="form-dimensions-input__length"
-			/>
-			<FormTextInput
-				name="width"
-				placeholder={ translate( 'W' ) }
-				type="number"
-				value={ dimensions && dimensions.width || '' }
-				onChange={ onChange }
-				className="form-dimensions-input__width"
-			/>
-			<FormTextInputWithAffixes
-				name="height"
-				placeholder={ translate( 'H' ) }
-				suffix={ unit }
-				type="number"
-				noWrap={ noWrap }
-				value={ dimensions && dimensions.height || '' }
-				onChange={ onChange }
-				className="form-dimensions-input__height"
-			/>
-		</div>
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { getDimensionsUnitSetting } from 'woocommerce/state/sites/settings/products/selectors';
+import { fetchSettingsProducts } from 'woocommerce/state/sites/settings/products/actions';
+
+class FormDimensionsInput extends Component {
+
+	static propTypes = {
+		className: PropTypes.string,
+		dimensions: PropTypes.shape( {
+			width: PropTypes.string,
+			height: PropTypes.sting,
+			length: PropTypes.string,
+		} ),
+		onChange: PropTypes.func.isRequired,
+		noWrap: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		value: '',
+		className: '',
+		onChange: noop,
+		noWrap: false,
+	}
+
+	componentDidMount() {
+		const { siteId } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchSettingsProducts( siteId );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.siteId !== this.props.siteId ) {
+			this.props.fetchSettingsProducts( newProps.siteId );
+		}
+	}
+
+	render() {
+		const { className, noWrap, dimensions, onChange, translate, dimensionsUnit } = this.props;
+		const classes = classNames( 'form-dimensions-input', className, { 'no-wrap': noWrap } );
+
+		return (
+			<div className={ classes }>
+				<FormTextInput
+					name="length"
+					placeholder={ translate( 'L' ) }
+					type="number"
+					value={ dimensions && dimensions.length || '' }
+					onChange={ onChange }
+					className="form-dimensions-input__length"
+				/>
+				<FormTextInput
+					name="width"
+					placeholder={ translate( 'W' ) }
+					type="number"
+					value={ dimensions && dimensions.width || '' }
+					onChange={ onChange }
+					className="form-dimensions-input__width"
+				/>
+				<FormTextInputWithAffixes
+					name="height"
+					placeholder={ translate( 'H' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'h' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).